docs/ww3_doxy_tmpl.md

```diff
@@ -0,0 +1,68 @@
+# WAVEWATCH III Doxygen Header Reference
+<br>
+
+_Doxygen markup headers to be placed directly above the respective Fortran code units._
+
+### FILE
+```
+!> @file  <file name>
+!> @brief  <one sentence description>
+!> 
+!> @author <author name> @date <dd-Mon-yyyy>
+```
+
+### MODULE
+```
+!> @brief <one sentence description>   # contents of _1. Purpose_ from original header
+!> 
+!> @details <extended description>
+!>
+!> @param <module var 1>               # ALL module variables documented
+!> ...
+!> @param <module var N>
+!> @author <author name> @date <dd-Mon-yyyy>
+```
+
+### PROGRAM
+```
+!> @brief  <one sentence description>      # contents of _1. Purpose_ from original header
+!>
+!> @details <extended description>          # contents of _2. Method_ from original header
+!> 
+!> @author <author name>  @date <dd-Mon-yyyy>
+```
+
+### SUBROUTINE
+```
+!> @brief  <one sentence description>       # contents of _1. Purpose_ from original header
```

Comment:
On subroutines and functions, the brief tag is not needed. The first sentence is automatically used as the brief description.

Again, this is to make the documentation less cluttered, and make it easier for programmers to properly document their functions without extra work.

So for functions and subroutines, we don't use either the brief or the details tags:

```
!>    This subrountine is to extract arbitrary size values from a
!>    packed bit string, right justifying each value in the unpacked
!>    array without skip and interations.
!>
!>    This should be used when input array IN has only one element. If IN
!>    has more elements, use G2_SBYTESC().
!>
!>    @param[in] IN Array input.
!>    @param[out] IOUT Unpacked array output.
!>    @param[in] ISKIP Initial number of bits to skip.
!>    @param[in] NBYTE Number of bits of each integer in IN to take.
!>
!>    @author Stephen Gilbert @date 2004-04-27
!>

SUBROUTINE G2_GBYTEC(IN,IOUT,ISKIP,NBYTE)
  character*1 in(*)
  integer iout(*)
  CALL G2_GBYTESC(IN,IOUT,ISKIP,NBYTE,0,1)
  RETURN
END SUBROUTINE G2_GBYTEC

```

Reply:
I left the brief in on subroutines/functions for the same reason I had left the details tag in, to me they add some structure when scanning through the code.  They are not required though so I could easily understand removing them.  I'll leave this call up to the code managers.

Reply:
@MatthewMasarik-NOAA I'll support whatever you decide is best.

Reply:
This decision has already been made and is working well. Is there some reason you guys want to be different from the other doxygen projects at NOAA?

Remove the unnecessary elements. Programmers will not thank you for adding extra requirements, which are not real. And having your programmers, and all future programmers, deal with the brief and details tag is just extra work and extra difference from the way the rest of the EMC are doing documentation.

Reply:
@edwardhartnett  I didn't know there was a NOAA wide mandate about use of doxygen. For WW3 we are really happy to start including doxygen coding updates and are super appreciative of the hard work @MatthewMasarik-NOAA has done to create these templates for WW3 to use.

Reply:
I have a meeting already scheduled with Ali this afternoon that will give us a chance to discuss offline.

Reply:
@edwardhartnett @JessicaMeixner-NOAA I'm putting together a new PR that will take these recommendations into consideration.  It will be posted this afternoon.

Reply:
Posted:  PR #593.